TypeCobol/Compiler/Diagnostics/CodeElementCheckers.cs

```diff
@@ -407,6 +407,58 @@ public static void OnCodeElement(StopStatement statement, CodeElementsParser.Sto
         }
     }
 
+    internal class ReferenceModifierChecker
+    {
+        private static Diagnostic CreateDiagnostic(string message, IParseTree location)
+        {
+            var position = ParseTreeUtils.GetFirstToken(location).Position();
+            return new Diagnostic(MessageCode.SyntaxErrorInParser, position, message);
+        }
+
+        private static void AddError(ParserRuleContextWithDiagnostics context, string message)
+        {
+            context.AttachDiagnostic(CreateDiagnostic(message, context));
+        }
+
+        public static void Check(ReferenceModifier referenceModifier, CodeElementsParser.ReferenceModifierContext context)
+        {
+            if (referenceModifier.LeftmostCharacterPosition == null)
+            {
+                AddError(context, "The position modifier must be defined.");
```

Comment:
This is not a "position modifier", I propose this message instead: "Left-most position of a reference modifier is required."